library/std/src/sys_common/locks/mod.rs

```diff
@@ -0,0 +1,27 @@
+cfg_if::cfg_if! {
+    if #[cfg(any(
+        target_os = "android",
+        target_os = "dragonfly",
+        all(target_os = "emscripten", target_feature = "atomics"),
+        target_os = "freebsd",
+        target_os = "hermit",
+        target_os = "linux",
+        target_os = "openbsd",
+        all(any(target_arch = "wasm32", target_arch = "wasm64"), target_feature = "atomics"),
+    ))] {
+        mod futex_mutex;
+        mod futex_rwlock;
+        mod futex_condvar;
+        pub(crate) use futex_mutex::Mutex;
+        pub(crate) use futex_rwlock::RwLock;
+        pub(crate) use futex_condvar::Condvar;
+    } else if #[cfg(target_os = "fuchsia")] {
+        mod futex_rwlock;
+        mod futex_condvar;
+        pub(crate) use crate::sys::locks::Mutex;
+        pub(crate) use futex_rwlock::RwLock;
+        pub(crate) use futex_condvar::Condvar;
+    } else {
+        pub(crate) use crate::sys::locks::{Mutex, RwLock, Condvar};
+    }
+}
```

Comment:
This probably isn't the right place either because sys_common [shouldn't](https://github.com/rust-lang/rust/blob/master/src/tools/tidy/src/pal.rs#L60) contain platform cfg() stuff even though tidy currently doesn't complain about it either.

I was going to suggest moving it to `sys::common` instead but its mod.rs does say

https://github.com/rust-lang/rust/blob/cd128880b13bc9f7a919567926e801c8a1487cdb/library/std/src/sys/common/mod.rs#L1-L5

So the #[path] seems to be the approved way already.

Reply:
Since this concerns other code like the thread parker and one-time initialization as well, I suggest we discuss this in #84187 (the tracking issue for that guideline).

Reply:
I don't want to devalue the contributions of anyone who has worked on libstd, but honestly, this mostly seems like it was one person's idea 2 years ago about what would be a good organization of stuff, tbh. Since then we've sprouted a lot more small niche mostly-tier-3 targets that are mostly just repeats of Linux or other POSIX platforms. I think we should revisit those guidelines heavily.